### DISCUSSION TITLE
`docker container inspect` returns exit code -1 with no output--why?

### DISCUSSION THREAD

**Discussion by karolz-ms** [docker/cli#5319] [Q&A]
I am working on .NET Aspire local application orchestrator; we use Docker CLI to orchestrate containers that support the local application run. Recently our users reported issues that we tracked down to `docker container inspect` command failing sometimes with no output and exit code -1 (Windows 11 fully patched, Docker Desktop latest version i.e. 4.33.1, running in WSL mode).

The typical use case during application startup is that the orchestrator creates and starts a bunch of containers, in fairly rapid succession, and then inspect each one 2-3 times, also fairly quickly (say within a few seconds from the launch). The exact command issued is

```shell
docker container inspect --format {{ json . }} <full container ID>
```

The problem occurs during second or third invocation (never first--that always succeeds). The invocation ends very quickly (couple milliseconds time), as if the CLI did not even attempt to call the daemon, and the exit code is -1 (captured via https://pkg.go.dev/os#ProcessState.ExitCode ). 

Any ideas why this might be occurring? E.g. does the CLI employ any sort of client-side throttling?

@savannahostrowski @laurazard @hades200082 @Alirexaa  FYI

P.S. I was never able to reproduce the problem myself, but some .NET Aspire community members (@hades200082 @Alirexaa ) have encountered this repeatedly.

The related .NET Aspire issue: https://github.com/dotnet/aspire/issues/5109

**Comment by thaJeztah** [comment]
At a quick glance, the only location I see `-1` being asigned to that field is in `container.State.Wait`, which sets it to `-1` if the context (request) either timed out or was cancelled; https://github.com/moby/moby/blob/ae1c95ccc7389bba4ef0f9620e450deacd08ea3d/container/state.go#L215-L227

```go
	go func() {
		select {
		case <-ctx.Done():
			// Context timeout or cancellation.
			resultC <- StateStatus{
				exitCode: -1,
				err:      ctx.Err(),
			}
			return
		case status := <-waitC:
			resultC <- status
		}
	}()
```

**Comment by danegsta** [comment]
I'm also working on the .NET Aspire orchestrator and a few times now I've noticed random errors related to `docker container create` returning a -1 exit code. Importantly, the container does successfully get created when this happens, despite the CLI command exit code indicating an error. I'm not seeing any output on stdio (stdout or stderr) when this happens. Based on the logging when this happens, there's less than 10 milliseconds spent running the create command, so it basically immediately exits when this occurs.

**Reply by danegsta** [reply]
Anecdotally, I think there are a few other issues I've seen but don't have logs for that make me think other commands have been returning -1 too; for example, I've seen `docker network connect` fail, but on retry, I get errors that the container is already connected.
